File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> hashmap;
        
        for(int i = 0; i < nums.size(); i++){
            int compliment = target - nums[i];
            if(hashmap.find(compliment) != hashmap.end()){
                return {hashmap[compliment], i};
            }
            hashmap.insert({nums[i], i});
        }
        
        return {};
    }
};

File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hashmap = {} # val : idx

        for i in range(len(nums)):
            diff = target - nums[i]
            if diff in hashmap:
                return [hashmap[diff], i]
            hashmap[nums[i]] = i

        return []
        